src/include/brain/workload/augmentedNN.h

```diff
@@ -0,0 +1,108 @@
+//===----------------------------------------------------------------------===//
+//
+//                         Peloton
+//
+// augmentedNN.h
+//
+// Identification: src/include/brain/workload/augmentedNN.h
+//
+// Copyright (c) 2015-2018, Carnegie Mellon University Database Group
+//
+//===----------------------------------------------------------------------===//
+
+#pragma once
+#include <numeric>
+#include <string>
+#include "brain/workload/base_tf.h"
+
+namespace peloton {
+namespace brain {
+
+template <typename Type>
+class TfSessionEntityInput;
+template <typename Type>
+class TfSessionEntityOutput;
+
+using TfFloatIn = TfSessionEntityInput<float>;
+using TfFloatOut = TfSessionEntityOutput<float>;
+
+/**
+ * AugmentedNN is used to predict selectivity or cardinality
+ * for range predicates, e.g, 
+ *   SELECT * FROM table WHERE c1 >= l1 AND c1 <= u1
+ *                         AND c2 >= l2 AND c2 <= u2
+ *                         AND ... 
+ * Input is [l1, u1, l2, u2, ...]

```

Comment:
From the tests, it looks like right now we're doing only one pair of predicates, just `[l1, u1]`, right?

Reply:
Right.